### ISSUE TITLE
Mapper can influence results ?

### ISSUE THREAD

**Issue by ptranvan** [PapenfussLab/gridss#330]
Hi @d-cameron,

I was wondering if different mappers could influence a lot the results. I am using bwa mem with default options.
Did you tried some options or different mappers ?

Thanks and have a good day.

**Comment by d-cameron** [comment]
There are two areas in which a read aligner is used:
1) Initial alignment of the input files (input to GRIDSS)
2) Split read identification from soft-clipped reads (GRIDSS-internal)

I found comparable performance on step 2 with bwa mem and bowtie2. bwa mem is the default as that has the largest install base thus is the least problematic dependency to have.

Step 2 is required to the run on the input files as aligners such as bowtie2 do not report split read alignment. In theory it's optional for bwa mem but I still run it as a) GRIDSS aims to be as aligner-agnostic as possible, and b) feeding soft-clipping reads back to bwa mem will give you additional split read alignments.

The step 2 aligner can be changed for the input and assembly (independently) by editing your `gridss.sh` driver script and adding `ALIGNER_COMMAND_LINE` arguments to the `gridss.SoftClipToSplitReads` command. E.g. to run bowtie2 you would add
```
ALIGNER_COMMAND_LINE=null
ALIGNER_COMMAND_LINE=
ALIGNER_COMMAND_LINE=
ALIGNER_COMMAND_LINE=
ALIGNER_COMMAND_LINE=

BOWTIE2_COMMAND_LINE = ImmutableList.of("bowtie2", "--threads", "%3$d", "--local", "--mm", "--reorder", "-x", "%2$s", "-U", "%1$s");
```


I have not extensively benchmarked the optimal aligner for step 1. Complete realignment of the sample is typically not a viable option for SV caller unless the project is a explicitly an SV project.

**Comment by d-cameron** [comment]
TLDR: yes, GRIDSS assumes aligner mapq is accurate and different aligners will give different results.
